python/lsst/afw/table/Catalog.i

```diff
@@ -255,7 +255,8 @@ public:
             # self._columns is created the when self.columns is accessed -
             # looking for it in self.columns below would trigger infinite
             # recursion.
-            if name == "_columns":
+            # Test for "this" as a temporary workaround for ticktet 7377
```

Comment:
I'm fairly certain it's coming from Swig: I think it must be trying to get it as an attribute before it exists, while wrapping that in a `try/except` block.  But that only works if `AttributeError` is actually raised when it doesn't exist (as opposed to infinite recursion).

It might be a good idea to look at the Swig generated code (if we can get enough of a traceback to find it) to confirm that this is the case, but we could defer that to DM-7377.